### DISCUSSION TITLE
Supbase storage client [WIP edition] for Golang

### DISCUSSION THREAD

**Discussion by NikhilCodes** [supabase/supabase#4486] [Show and tell]
Not sure if its the right place but I'll ask anyway.

I would like to work on supabase storage client for Golang.

Some initial work sample can be found here,
[https://github.com/NikhilCodes/storage-go](https://github.com/NikhilCodes/storage-go)

- [x] Bucket list
- [x] Create bucket
- [x] Delete bucket
- [x] Edit bucket
- [x] File Upload utils
- [x] ...

**Comment by kiwicopple** [comment]
Hey @NikhilCodes - this looks great! If you want, I can move this over to the supabase-community?
https://github.com/supabase-community/

If you're happy with that, can you reach out on email (copple at supabase dot io) or discord?

**Reply by NikhilCodes** [reply]
Hey @kiwicopple  I added file upload utilities just now, which includes:

 - Upload file
 - Update file
 - Move file
 - Create Signed url
 - Get Public url
 - Remove file
 - List files
 
Do let me know what I need to do to transition it into https://github.com/supabase-community/
